src/test/java/org/jvnet/hudson/test/RealJenkinsRuleHttpsTest.java

```diff
@@ -0,0 +1,46 @@
+package org.jvnet.hudson.test;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class RealJenkinsRuleHttpsTest {
+    private static final Logger LOGGER = Logger.getLogger(RealJenkinsRuleHttpsTest.class.getName());
+
+    @Rule
+    public final RealJenkinsRule rr = new RealJenkinsRule().https();
+
+    @Rule
+    public InboundAgentRule iar = new InboundAgentRule();
+
+    @Before
+    public void setUp() throws Throwable {
+        rr.startJenkins();
+    }
+
+    @Test
+    public void runningStepAndUsingHtmlUnit() throws Throwable {
+        // We can run steps
+        rr.runRemotely(RealJenkinsRuleHttpsTest::log);
+        // replica1 directly
+        try (var wc = rr.createWebClient()) {
+            wc.getPage(rr.getUrl());
+        }
+    }
+
+    @Test
+    public void inboundAgent() throws Throwable {
+        var options = InboundAgentRule.Options
+                .newBuilder()
+                .name("remote")
+                .webSocket()
+                .color(PrefixedOutputStream.Color.YELLOW);
+        iar.createAgent(rr, options.build());
+    }
+
+    private static void log(JenkinsRule r) throws IOException {
+        LOGGER.info("Running on " + r.getURL().toExternalForm());
```

Comment:
Could strengthen this slightly to return the URL and show that it is `https`.

Reply:
Prints something like
```
[INFO] [stdout] 2024-10-10 12:53:16.687+0000 [id=94]    INFO    o.j.h.t.RealJenkinsRuleHttpsTest#log: Running on https://localhost:54912/jenkins/
```

Reply:
Yes, I just meant that the test JVM could assert that.